tests/metatables.cpp

```diff
@@ -41,3 +41,40 @@ TEST(Metatables, WritingMetatableObjects)
 
     EXPECT_EQ(5, context.readVariable<Foo>("foo").value);
 }
+
+TEST(Metatables, customEq)
+{
+   struct Foo { int value = 0; bool operator==(const Foo& rhs) { return rhs.value == this->value; }; };
+   struct Bar { int value = 0; };
+
+   LuaContext context;
+
+   context.registerFunction("__eq", &Foo::operator==);
```

Comment:
I'd like `registerFunction` to be named differently for this purpose, so that we have the opportunity to move `__eq` (and `__tostring` below) into the metatable if we ever figure out how.